index.js

```diff
@@ -386,6 +410,122 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread, callback) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+    var deepThread = pluckRecursively(thread, 'related')[0]
+
+    var threadRevisionLogCB = multicb({pluck: 1})
+    deepThread
+    .filter(function(relatedMsg) { return isaRevisionTo(relatedMsg, thread) })
+    .forEach(function(edit) {
+      exports.createRevisionLog(ssb, edit, threadRevisionLogCB())
+    })
+    
+    threadRevisionLogCB(function(err, revLogs) {
+      if (err) callback(err)
+      const connectedLogs = revLogs
+        .filter(function(revLog) { 
+          // filter down to the log that links to this thread, by finding the log
+          // whose messages point to this thread as root 
+          // we should only need to test one message if createRevisionLog is
+          // passing tests
+          return revLog[revLog.length-1].key === thread.key
+        })
+
+      // the longest such log is the one that contains all of the revisions
+      const logLengths = connectedLogs.map(function(log) { return log.length })
+      const thisLog = connectedLogs.find(function(log) {
+                        return log.length === Math.max.apply(Math, logLengths)
+                      })
+      callback(null, thisLog)
+    })
+    
+  }
+            
+
+  if (!thread.hasOwnProperty('related')) { 
+    // if the thread doesn't have its related objects,fetch them
+    ssb.relatedMessages({id: thread.key, count: true}, function(err, enrichedThread) {
+      if (err) callback (err)
+      // if still no related objects
+      else if (!enrichedThread.hasOwnProperty('related')) callback(null, [])
+      else collectRevisions(enrichedThread, callback)
+    })
+  } else {
+    collectRevisions(thread, callback)
+  }  
+}
+
+exports.getLatestRevision = function (ssb, thread, callback) {  
+  var msg = thread
+  if (!msg.hasOwnProperty('related')) {
+    ssb.relatedMessages({id: msg.key, count: true}, function(err, richMsg) {
+      if (err) callback (err)
+      // if still no related objects
+      else if (!richMsg.hasOwnProperty('related')) { callback(null, richMsg) }
+      else { traverseRelatedEdits(richMsg, callback) }
+    })
+  } else if (!msg.hasOwnProperty('related')) {
+    callback(null, msg)
+  } else {
+    traverseRelatedEdits(msg, callback)
+  }
```

Comment:
Yeah, this is harder than it should be, because the old linking structure. Once `revisionRoot` is in place, the way this will work is by getting the most recent `.related` that points to the target message as a `revisionRoot`. This will be the new filter:

```js
.filter(function (rev) {
  return rev.value.content.type === 'post-edit'
    && mlib.relationsTo(rev, thread).indexOf('revisionRoot') !== -1)
})
```

If it's cool with you, I can fork these PRs and make the updates I'm describing. We're almost there, and I don't want to keep tasking you with updates when the turn-around is so long.